dvc/remote/local/__init__.py

```diff
@@ -566,3 +601,73 @@ def unprotect(path_info):
     @staticmethod
     def protect(path_info):
         os.chmod(path_info.path, stat.S_IREAD | stat.S_IRGRP | stat.S_IROTH)
+
+    @classmethod
+    def _append_unpacked_suffix(cls, string):
+        return string + cls.UNPACKED_DIR_SUFFIX
+
+    def _get_unpacked_dir_path_info(self, checksum):
+        unpacked_dir_info = self.checksum_to_path_info(checksum)
+        unpacked_dir_info.path = self._append_unpacked_suffix(
+            unpacked_dir_info.path
+        )
+
+        return unpacked_dir_info
+
+    def _path_info_changed(self, path_info):
+        if self.exists(path_info) and self.state.get(path_info):
+            return False
+        return True
+
+    def _update_unpacked_dir(self, checksum, progress_callback=None):
+        unpacked_dir_info = self._get_unpacked_dir_path_info(checksum)
+
+        if not self._path_info_changed(unpacked_dir_info):
+            return
+
+        self.remove(unpacked_dir_info)
+
+        try:
+            dir_info = self.get_dir_cache(checksum)
+            self._create_unpacked_dir(
+                checksum, dir_info, unpacked_dir_info, progress_callback
+            )
+        except Exception:
+            logger.warning(
+                "Could not create '{}'".format(unpacked_dir_info.path)
+            )
+
+            self.remove(unpacked_dir_info)
+
+    def _create_unpacked_dir(
+        self, checksum, dir_info, unpacked_dir_info, progress_callback=None
+    ):
+        self.makedirs(unpacked_dir_info)
+        for index, entry in enumerate(dir_info):
+            entry_cache_info = self.checksum_to_path_info(
+                entry[self.PARAM_CHECKSUM]
+            )
+            relpath = entry[self.PARAM_RELPATH]
+
+            unpacked_entry_info = copy(entry_cache_info)
+            unpacked_entry_info.path = self.ospath.join(
+                unpacked_dir_info.path, relpath
+            )
+            self.link(entry_cache_info, unpacked_entry_info, "hardlink")
+
+            if progress_callback:
+                progress_callback.update("Creating unpacked dir.")
+        self.state.save(unpacked_dir_info, checksum)
+
+    def _changed_unpacked_dir(self, checksum):
+        status_unpacked_dir_info = self._get_unpacked_dir_path_info(checksum)
+
+        return not self.state.get(status_unpacked_dir_info)
+
+    def extract_used_local_checksums(self, cinfos):
+        used = super(RemoteLOCAL, self).extract_used_local_checksums(cinfos)
+        unpacked_dir = set()
```

Comment:
We already have the notion of unpacked dirs in the RemoteBase (which I like and it makes total sense to me), let's then not hide this method in RemoteLOCAL and keep it in RemoteBASE, so we don't have to overwrite this.